controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -311,18 +312,9 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
                 return CompletableFuture.completedFuture(AddSubscriberStatus.Status.STREAM_NOT_FOUND);
             } else {
                 // 2. get subscribers data
-                return Futures.exceptionallyExpecting(streamMetadataStore.getSubscriber(scope, stream, newSubscriber, context, executor),
-                     e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, null)
-                    .thenCompose(subscribersData -> {
-                    //4. If SubscriberRecord does not exist create one...
-                    if (subscribersData == null) {
-                        return streamMetadataStore.createSubscriber(scope, stream, newSubscriber, context, executor)
-                                .thenApply(v -> AddSubscriberStatus.Status.SUCCESS);
-                    } else {
-                        return CompletableFuture.completedFuture(AddSubscriberStatus.Status.SUBSCRIBER_EXISTS);
-                    }
-                    })
-                    .exceptionally(ex -> {
+                return streamMetadataStore.createSubscriber(scope, stream, newSubscriber, generation, context, executor)
```

Comment:
question: does create subscriber update the entry if it already existed, but with an older generation?

Reply:
If entry already exists, its generation is updated only if the (API generation) > stored generation and not otherwise.